codice/ddf
# DDF-2863 Added ability to set servlet filter initialization parameters

#### What does this PR do?
This PR
- Adds an experimental `SecurityService` marker interface.
- Updates the `DelegateServletFilter` to
  - gather services that implement the `SecurityService` interface instead of the `javax.servlet.Filter` interface.
  - only `init` each filter once instead of on every `doFilter` call.
- Updates current filters to
  - implement the `SecurityService` interface instead of the `javax.servlet.Filter` interface.
  - use the Pax Web Extender filter registration notation. See [Pax Web Extender Whiteboard](https://ops4j1.jira.com/wiki/display/ops4j/Pax+Web+Extender+-+Whiteboard), [`org.ops4j.pax.web.extender.whiteboard.internal.tracker.ServletTracker`](https://github.com/ops4j/org.ops4j.pax.web/blob/web-6.0.3/pax-web-extender-whiteboard/src/main/java/org/ops4j/pax/web/extender/whiteboard/internal/tracker/ServletTracker.java), and [`org.ops4j.pax.web.extender.whiteboard`](https://github.com/ops4j/org.ops4j.pax.web/blob/web-6.0.3/pax-web-extender-whiteboard/src/main/java/org/ops4j/pax/web/extender/whiteboard/ExtenderConstants.java).
  - remove servlet-specific properties because the `DelegateServletFilter` adds the filters to every servlet, so these properties we never being used.
- Removes the `ServiceComparator` and `SortedServiceList` classes from the `platform-filter-delegate` module because they are not needed.

#### Who is reviewing it?
@mcalcote @vinamartin @josephthweatt @alexaabrd 
#### Select relevant component teams: 
@codice/security @stustison
#### Choose 2 committers to review/merge the PR. 
@clockard
@pklinef
@stustison
#### How should this be tested?
- Confirm unit tests and itests succeed.
- Add
  ```
  log4j2.logger.org_codice_ddf_platform_filter_delegate_delegateservletfilter.level = DEBUG
  log4j2.logger.org_codice_ddf_platform_filter_delegate_delegateservletfilter.name = 
  org.codice.ddf.platform.filter.delegate.DelegateServletFilter
  ```
  to `etc/org.ops4j.pax.logging.cfg` before starting DDF. Confirm log statements for initializing `SecurityFilter`s in the `DelegateServletFilter` are correct when DDF is running.
- Confirm manual install succeeds.
#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-2863](https://codice.atlassian.net/browse/DDF-2863)
#### Screenshots (if appropriate)
#### Checklist:
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
